src/engine/Settings.cs

```diff
@@ -395,7 +455,106 @@ private void CopySettings(Settings settings)
             if (!property.CanWrite)
                 continue;
 
-            property.SetValue(this, property.GetValue(settings));
+            // Since the properties we want to copy are SettingValue generics we use the IAssignableSetting
+            // interface and AssignFrom method to convert the property to the correct concrete class.
+            var setting = property.GetValue(this) as IAssignableSetting;
+
+            setting?.AssignFrom(property.GetValue(settings));
+        }
+    }
+
+    /// <summary>
+    ///   Wrapper class for settings options containing the value and a delegate that provides a callback for
+    ///   when the value is changed.
+    /// </summary>
+    public class SettingValue<TValueType> : IAssignableSetting
```

Comment:
I think this should be put in a separate file as this is a bit long class.

The interface could be defined also in that file, then, or in a new file.